lib/stdlib/src/timer.erl

```diff
@@ -26,129 +26,192 @@
 	 cancel/1, sleep/1, tc/1, tc/2, tc/3, now_diff/2,
 	 seconds/1, minutes/1, hours/1, hms/3]).
 
--export([start_link/0, start/0, 
-	 handle_call/3,  handle_info/2,  
+-export([start_link/0, start/0,
+	 handle_call/3,  handle_info/2,
 	 init/1,
 	 code_change/3, handle_cast/2, terminate/2]).
 
-%% internal exports for test purposes only
--export([get_status/0]).
-
 %% types which can be used by other modules
 -export_type([tref/0]).
 
-%% Max
--define(MAX_TIMEOUT, 16#0800000).
--define(TIMER_TAB, timer_tab).
--define(INTERVAL_TAB, timer_interval_tab).
+%% Max value for a receive's after clause.
+-define(MAX_RECEIVE_AFTER, 16#ffffffff).
+
+%% validations
+-define(valid_time(T), is_integer(T), T >= 0).
+-define(valid_mfa(M, F, A), is_atom(M), is_atom(F), is_list(A)).
 
 %%
 %% Time is in milliseconds.
 %%
--opaque tref()    :: {integer(), reference()}.
+-opaque tref()    :: {type(), reference()}.
+-type type()      :: 'once' | 'interval'.
 -type time()      :: non_neg_integer().
 
 %%
 %% Interface functions
 %%
 -spec apply_after(Time, Module, Function, Arguments) ->
-                         {'ok', TRef} | {'error', Reason} when
-      Time :: time(),
-      Module :: module(),
-      Function :: atom(),
-      Arguments :: [term()],
-      TRef :: tref(),
-      Reason :: term().
-
-apply_after(Time, M, F, A) ->
-    req(apply_after, {Time, {M, F, A}}).
-
--spec send_after(Time, Pid, Message) -> {'ok', TRef} | {'error', Reason} when
-      Time :: time(),
-      Pid :: pid() | (RegName :: atom()),
-      Message :: term(),
-      TRef :: tref(),
-      Reason :: term().
-send_after(Time, Pid, Message) ->
-    req(apply_after, {Time, {?MODULE, send, [Pid, Message]}}).
-
--spec send_after(Time, Message) -> {'ok', TRef} | {'error', Reason} when
-      Time :: time(),
-      Message :: term(),
-      TRef :: tref(),
-      Reason :: term().
+	    {'ok', TRef} | {'error', Reason}
+  when Time :: time(),
+       Module :: module(),
+       Function :: atom(),
+       Arguments :: [term()],
+       TRef :: tref(),
+       Reason :: term().
+apply_after(0, M, F, A)
+  when ?valid_mfa(M, F, A) ->
+    do_apply({M, F, A});
+apply_after(Time, M, F, A)
+  when ?valid_time(Time),
+       ?valid_mfa(M, F, A) ->
+    {ok, Pid} = ensure_started(),
+    try
+	erlang:send_after(Time, Pid, {apply_once, {M, F, A}})
+    of
+	TRef ->
+	    {ok, {once, TRef}}
+    catch
+	error:badarg ->
+	    {error, badarg}
+    end;
+apply_after(_Time, _M, _F, _A) ->
+    {error, badarg}.
+
+-spec send_after(Time, Pid, Message) -> {'ok', TRef} | {'error', Reason}
+  when Time :: time(),
+       Pid :: pid() | (RegName :: atom()),
+       Message :: term(),
+       TRef :: tref(),
+       Reason :: term().
+send_after(0, PidOrRegName, Message)
+  when is_pid(PidOrRegName);
+       is_atom(PidOrRegName) ->
+    PidOrRegName ! Message;
+send_after(0, Dest = {RegName, Node}, Message) % Undocumented; for backwards compatibility.
```

Comment:
I don't see any reason to not document the `{Name, Node}` and remove the comment. Do you?

Reply:
It's not working with `send_interval` and up to now I thought this couldn't be done, but I just realized that it actually is possible now.
I also see that there is actually a slight (unintended, @juhlig?) change in behavior in `send_interval`.
I'll see to it.

Reply:
@Maria-12648430... so is it still Easter or what? ^^;;;

Reply:
@juhlig yes 😁 (Hint: check your mail)

Reply:
Added to docs in last commit, also changed `send_interval` to match.

Just noticed that I forgot to remove the comment, though. Will do that later.

Now, the only functions that do not accept `{RegName, Node}` tuples are `exit_after` and `kill_after`, because `exit/2`  only accepts pids, and TBH I don't know how to resolve the tuple to a pid 😅

Reply:
... or even do the `exit` itself via `rpc:cast/4`?

Reply:
Hm, now one thing came to mind regarding the original question about documenting `{RegName, Node}`: `Node` will be automatically connected by the send operation. This is no change of behavior, though, it would have done that before, too.